### PR TITLE
Fix placement of `attachments` array in interaction response payload

### LINKED ISSUE #218: Upcoming API change to interaction response attachment handling
### Description

The API currently reads attachment data for interaction responses from `payload["attachments"]`, which is also where we put attachment data. This, however, does not match the API documentation, which correctly specifies `payload["data"]["attachments"]`; the API side of things is subject to change soon:

> We recently discovered our API had a bug where attachments wasn't nested under data - our API was reading from payload_json["attachments"] instead of payload_json["data"]["attachments"]. This isn't what we have documented, and we're considering it an implementation bug (it was introduced when I shipped multi-upload support a few weeks ago). 

(full quote here: https://discord.com/channels/808030843078836254/847513235536216085/922602341151162418, private channel)

disnake is affected by this change as well:
https://github.com/DisnakeDev/disnake/blob/9bb3821fdc458b1a5b729685e22c1887044482d0/disnake/webhook/async_.py#L414-L419

### Impact
Since API v10 isn't a thing yet and providing an `attachments` value is hence not required, this only affects attachment descriptions ~~and removing existing files on edit~~, and **not** plain file upload, as far as I can tell.

Coordinating this might be difficult, since there will likely be no transitional phase where both options work, so we should have a PR and a bugfix release ready - the change will happen "shortly after we come back from our break in the new year" (also taken from the quote mentioned above), which according to an announcement in games-lab will end January 3rd.

---

*Edit:* this does not in fact affect removing files on edit, since the endpoint currently doesn't even support that (see https://github.com/discord/discord-api-docs/discussions/3335)

### PR DESCRIPTION
**Pull request by shiftinv**
## Summary

This PR moves the `attachments` list into the `data` object in interaction response payloads, see #218 for details.
The important part of this PR is `create_interaction_response` / e752f2d0e7d625dfa6233cbb66c5ba2dd06b5088, where attachments are now modified *before* the payload is moved into `data`.

Ideally this should only be merged shortly before the API change is deployed, an exact date for that isn't known yet, afaik.

Fixes #218.

## Checklist

- [x] If code changes were made, then they have been tested
    - [ ] I have updated the documentation to reflect the changes
    - [x] I have formatted the code properly by running `pre-commit run --all-files`
- [x] This PR fixes an issue
- [ ] This PR adds something new (e.g. new method or parameters)
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: disnake/http.py [hunk 1]
```diff
@@ -111,20 +111,15 @@ async def json_or_text(response: aiohttp.ClientResponse) -> Union[Dict[str, Any]
     return text
 
 
-def to_multipart(payload: Dict[str, Any], files: Iterable[File]) -> List[Dict[str, Any]]:
-    # NOTE: this method modifies the provided `payload` and `payload["attachments"]` collections
+def set_attachments(payload: Dict[str, Any], files: Sequence[File]) -> None:
+    """
+    Updates the payload's attachments list based on the provided files
+
+    note: this method modifies the provided `payload` and `payload["attachments"]` collections
```
Comment (EQUENOS) [inline disnake/http.py:118]: You probably wanted to type \`\`payload\`\` and \`\`payload["attachments"]\`\`, same thing in other docstrings.
Doesn't matter though, these functions aren't public anyways.

**Review by EQUENOS — APPROVED** [review]
Good stuff

**Comment by shiftinv** [comment]
This change appears to be live now, attachment descriptions don't on master anymore

**Comment by Victorsitou** [comment]
> This change appears to be live now.

Is it sending the interaction or it raises an error? (iirc the feature is still on beta testing)

**Pull request merged**